x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/inference/inference_usage.yml

```diff
@@ -0,0 +1,155 @@
+setup:
+  - requires:
+      cluster_features: "inference.semantic_text_usage"
+      reason: semantic text stats added to inference usage
+
+---
+"Test usage given default endpoints":
+
+  - do:
+      xpack.usage: {}
+
+  - match: { inference.available: true }
+  - match: { inference.enabled: true }
+  - length: { inference.models: 11 }
+
+  - match: { inference.models.0.service: "_all" }
+  - match: { inference.models.0.task_type: "CHAT_COMPLETION" }
+  - match: { inference.models.0.count: 0 }
+  - match: { inference.models.0.semantic_text: { field_count: 0, indices_count: 0, inference_id_count: 0} }
```

Comment:
Semantic text is not compatible with the `CHAT_COMPLETION` task type. Should the semantic_text field be optional in these stats objects and not reported for incompatible task types

Reply:
I have addressed this in https://github.com/elastic/elasticsearch/pull/135262/commits/809631b4946fb98c06f85f5d0e3a941fc279df1d